### ISSUE TITLE
[API] Rename #dtype to #backend or something like this

### ISSUE THREAD

**Issue by zverok** [v0dro/daru#140]
**Reason**: in many cases (like NMatrix), as well as intuitively, `container.dtype` reads like "type of data stored in this container" (like numbers or string or what). But Daru's `#dtype` means "storage backend", which is confusing.

**Comment by v0dro** [comment]
Agreed. How about `container`, or `data_store`? We'll also need to make some non-trivial changes in notebooks.
